clang/runtime/dpct-rt/include/math.hpp

```diff
@@ -510,6 +521,36 @@ inline unsigned vectorized_sum_abs_diff(unsigned a, unsigned b) {
   }
   return sum;
 }
+
+template <typename retT, typename AT, typename BT, class BinaryOperation>
+inline constexpr retT extend_binary(const AT a, const BT b,
+                                    const BinaryOperation binary_op,
+                                    bool need_sat = false) {
```

Comment:
related instructions include clamp, warp, etc., not just sat

Reply:
vadd/vsub/vabsdiff/amin/vmax can only have sat:
```
// 32-bit scalar operation, with optional secondary operation
vop.dtype.atype.btype{.sat}       d, a{.asel}, b{.bsel};
vop.dtype.atype.btype{.sat}.op2   d, a{.asel}, b{.bsel}, c;

// 32-bit scalar operation, with optional data merge
vop.dtype.atype.btype{.sat}  d.dsel, a{.asel}, b{.bsel}, c;

 vop   = { vadd, vsub, vabsdiff, vmin, vmax };
.dtype = .atype = .btype = { .u32, .s32 };
.dsel  = .asel  = .bsel  = { .b0, .b1, .b2, .b3, .h0, .h1 };
.op2   = { .add, .min, .max };
```

Reply:
But it's difficult to extend to other APIs, such as `vshl`